devlib/instrument/baylibre_acme.py

```diff
@@ -0,0 +1,424 @@
+        try:
+            # This will fail if there is not exactly one single value:
+            (value,) = {getattr(probes[p], property_name) for p in probes}
+        except ValueError:
+            msg = 'Probes have different values for {}.'
+            raise ValueError(msg.format(property_name))
+        return value
+
+    def setter(self, value):
+        # This setter is required for compliance with the inherited methods
+        if value is not None:
+            raise AttributeError("can't set attribute")
+
+    return property(getter, setter if none_accepted else None)
+
+class BaylibreAcmeInstrument(Instrument):
+
+    def __init__(self, iio_context=None, use_base_iio_context=False, probe_names=None):
+        if isinstance(probe_names, basestring):
+            probe_names = [probe_names]
+
+        self.iio_context = (iio_context if not use_base_iio_context
+                            else iio.Context(iio_context))
+
+        if probe_names is not None:
+            if len(probe_names) != len(self.iio_context.devices):
+                msg = ('There should be as many probe_names ({}) '
+                       'as detected probes ({}).')
+                raise ValueError(msg.format(len(probe_names),
+                                            len(self.iio_context.devices)))
+
+            probes = [IIOINA226Instrument(d, name)
+                      for d, name in zip(self.iio_context.devices, probe_names)]
+        else:
+            probes = [IIOINA226Instrument(d) for d in self.iio_context.devices]
+
+        self.probes = {p.nickname : p for p in probes}
+
+    # channel management
+
+    def list_channels(self):
+        # TODO: not sure what to do:
+        pass
+        #      or
+        # return {p: self.probes[p].list_channels() for p in self.probes}
+
+    @staticmethod
+    def add_channel():
+        pass # Channels are handled by the probes
+
+    # properties
+
+    sample_rate_hz = _BaylibreAcmePropertyFromProbes('sample_rate_hz', True)
+    oversampling_ratios_available = \
+            _BaylibreAcmePropertyFromProbes('oversampling_ratios_available')
+    integration_times_available = \
+            _BaylibreAcmePropertyFromProbes('integration_times_available')
+
+    # initialization and teardown
+
+    def setup(self, shunt_resistor,
+              integration_time_bus,
+              integration_time_shunt,
+              oversampling_ratio,
+              buffer_samples_count=None,
+              buffer_is_circular=False,
+              absolute_timestamps=False):
+
+        args = {k : v for k, v in locals().iteritems() if k != 'self'}
+
+        def pseudo_list(v, i):
+            try:
+                return v[i]
+            except TypeError:
+                return v
+
+        for i, p in enumerate(self.probes.values()):
+            p.setup(**{k : pseudo_list(v, i) for k, v in args.iteritems()})
+
+    def reset(self, probes=None, sites=None, kinds=None, channels=None):
```

Comment:
The breaks the API by adding `probes`; and, as mentioned above, probes == sites.